File: src/util/config/types/GeneralConfiguration.ts
import { Snowflake } from "@fosscord/util";

export class GeneralConfiguration {
	instanceName: string = "Fosscord Instance";
	instanceDescription: string | null = "This is a Fosscord instance made in the pre-release days";
	frontPage: string | null = null;
	tosPage: string | null = null;
	correspondenceEmail: string | null = null;
	correspondenceUserID: string | null = null;
	image: string | null = null;
	instanceId: string = Snowflake.generate();
}

File: extra/admin-api/Spacebar.ConfigModel/GeneralConfiguration.cs
using System.Text.Json.Serialization;

namespace Spacebar.ConfigModel;

public class GeneralConfiguration {
    [JsonPropertyName("instanceName")]
    public string InstanceName { get; set; } = "Spacebar Instance";

    [JsonPropertyName("serverName")]
    public string? ServerName { get; set; } = null;

    [JsonPropertyName("instanceDescription")]
    public string InstanceDescription { get; set; } = "This is a Spacebar instance made in the pre-release days";

    [JsonPropertyName("frontPage")]
    public string? FrontPage { get; set; } = null;

    [JsonPropertyName("tosPage")]
    public string? TosPage { get; set; } = null;

    [JsonPropertyName("correspondenceEmail")]
    public string? CorrespondenceEmail { get; set; } = null;

    [JsonPropertyName("correspondenceUserID")]
    public string? CorrespondenceUserID { get; set; } = null;

    [JsonPropertyName("image")]
    public string? Image { get; set; } = null;

    [JsonPropertyName("instanceId")]
    public string InstanceId { get; set; } = null!; // {get;set;}=Snowflake.generate();

    [JsonPropertyName("autoCreateBotUsers")]
    public bool AutoCreateBotUsers { get; set; } = false;
}